mso/datasource_mso_schema_template_contract.go

```diff
@@ -62,98 +68,69 @@ func dataSourceMSOTemplateContract() *schema.Resource {
 							Type:     schema.TypeString,
 							Computed: true,
 						},
+						"filter_type": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
 						"action": &schema.Schema{
 							Type:     schema.TypeString,
 							Computed: true,
 						},
 						"directives": {
-							Type:     schema.TypeList,
+							Type:     schema.TypeSet,
 							Elem:     &schema.Schema{Type: schema.TypeString},
 							Computed: true,
 						},
+						"priority": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"filter_relationships": {
+				Type:     schema.TypeMap,
```

Comment:
This need to be TypeList, aren't you mixing up filter_relationships and filter_relationship in the data source vs the resource?

Reply:
No we had the discussion regarding this. I originally got confused by this and introduced a change in (https://github.com/CiscoDevNet/terraform-provider-mso/commit/fdb2c1e082d8cdcbd16563146f5e41c8d98edc22) but that was wrong change. Occurred due to the confusing naming. What I should have done back then was introduce the filter_relationship type list attribute in the datasource, to be consistent with the resource. This change is the revert of that commit with the new attributes added to the filter_relationship (list type).